### ISSUE TITLE
question on using merge tensor (or Merge Layer)

### ISSUE THREAD

**Issue by hengck23** [fchollet/keras#5586]
I am trying to convert my tensorflow code to keras. I have a problem  about concatenation layer
![001](https://cloud.githubusercontent.com/assets/5112846/23545648/31bc1a5a-0037-11e7-9c39-caca0079b505.png)

In the construction of "dense-block2" of the diagram, my keras code gives the error " 'concat' mode can only merge layers with matching output shapes except for the concat axis. Layer shapes: [(None, 16, 16, 32), (None, 16, 16, 16)]". What is wrong?
(Note: I use the latest tensorflow 1.0 relase as backend. )

```
    #block1
    model.add(Convolution2D(32, 5, 5, subsample=(1, 1), border_mode='same', bias=False, W_regularizer=l2(decay)))
    model.add(BatchNormalization())
    model.add(Activation(relu))
    model.add(MaxPooling2D(pool_size=(2, 2), strides=(2,2), border_mode='same'))

    #block2
    for n in range(4):
        left  = model.output
        right = Convolution2D(16, 3, 3, subsample=(1, 1), border_mode='same', bias=False, W_regularizer=l2(decay))(left)
        right = BatchNormalization()(right)
        right = Activation(relu)(right)
        model.add(merge([left, right], mode='concat', concat_axis=1))
    model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2), border_mode='same'))

```

**Comment by hengck23** [comment]
i solved the issue.
It should have been : model.add(merge([left, right], mode='concat', concat_axis=3))
Also the tf.concat() in tensorflow has been changed.
